prow/githuboauth/githuboauth.go

```diff
@@ -54,13 +54,30 @@ type GitHubClientGetter interface {
 
 // OAuthClient is an interface for a GitHub OAuth client.
 type OAuthClient interface {
+	GetConfig() *oauth2.Config
 	// Exchanges code from GitHub OAuth redirect for user access token.
 	Exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error)
 	// Returns a URL to GitHub's OAuth 2.0 consent page. The state is a token to protect the user
 	// from an XSRF attack.
 	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
 }
 
+type RealOAuthClient struct {
```

Comment:
Why is this "real"?

Reply:
Got rid of this. It was "real" in contrast to the mock OAuthClient in the tests.